Add temperature dependant thermal expansion to tutorial

aeslaughter opened the pull request:
<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

aeslaughter committed:
Update step9 to use temperature dependant cte

(refs #4522)

(refs #4522)

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/352398/) on 67c32e6 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/13575/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/13575/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format bff2281da68ee2a727d8d6366d37b01ff0f2d09b`

aeslaughter committed:
Update steps 9 and 10 to use AD in TM

(refs #4522)

aeslaughter committed:
Update large media submodule

(refs #4522)

moosebuild commented on large_media:
```diff
@@ -1 +1 @@
-Subproject commit de952acdd891280ced118dae0c8beb20551ba4cd
+Subproject commit d75a1d198954cf4d33383bf7ebd25bd2f1e3fe09
```
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

moosebuild replied:
**Caution!** This contains a submodule update

aeslaughter commented:
@lindsayad If you have some time, can you take a look at step 9.

I updated everything that I can to AD, even tensor mechanics stuff. But, NEWTON doesn't seem to do all that great. Also, I switched to PJFNK with SMP/full=true and it fails completely.

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/352458/) on 84d15a5 wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/13575/site)

This comment will be updated on new commits.

aeslaughter committed:
Update for automatic scaling

(refs #4522)

lindsayad commented on tutorials/darcy_thermo_mech/doc/content/problem/step06.md:
```diff
@@ -58,15 +58,15 @@ C\left( \frac{\partial T}{\partial t} + \underbrace{\epsilon \vec{u}\cdot\nabla
+## Auto Variable Scaling
 
 To obtain an optimum numerical solution, the non-linear variables should be on the same
 scale. The condition number can be used to determine
```
Can be used to determine what?

lindsayad commented on tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i:
```diff
@@ -217,6 +211,8 @@
   dt = 0.25
   solve_type = PJFNK
+  automatic_scaling = true
+  compute_scaling_once = false
```
So we saw that this made the solve significaintly better. For further debugging I would be curious how this does with `-pc_type lu`. For those time steps that we saw taking a lot of linear iterations, I would be curious to know how Jacobian-free is doing at that point. I'm wondering whether at that point whether our Jacobian-free approximation is doing a poor job of representing the Jacobian. It would be good to run with `Outputs/print_linear_residuals=false -ksp_monitor_true_residual` and see whether the true residual is dropping. Does this problem get more and more non-linear over time?

lindsayad replied:
> Jacobian-free approximation is doing a poor job of representing the Jacobian

Sorry this would show up more in the non-linear iterations, not the linears. But you can see cases where the GMRES residuals drop but the true residuals do not; this usually indicates that your Jacobian-free process is non-linear, e.g. due to finite differencing of a noisy function or something. With good scaling though, that should be less of a concern.

Actually the most obvious explanation for a degrading linear solve would be if our off-diagonal terms are getting stronger, since we are currently only using the diagonals in the preconditioning matrix. Is that possibly happening? Won't happen in time for training, but we definitely want to figure out where our Jacobian bug is! Then we could run with NEWTON :smile:

aeslaughter replied:
Thanks for digging into this, I think what I have in this PR is good for the training. We will get to this example sometime on Tue., so you might be able to help with this section and could talk about solver options and such.

aeslaughter replied:
-pc_type lu works much better, it doesn't seem to get worse over time.

moosebuild commented:
All [jobs](https://civet.inl.gov/event/43148/) on 93085be : invalidated by @aeslaughter

BISON has been updated

aeslaughter committed:
Step 9 remove material property, it isn't supported yet, but things are ready when it does

(refs #4522)

aeslaughter commented:
@permcody Can you take a quick look, this is phase 1 of the changes that I need to get in. I am working on the stuff from our meeting.

aeslaughter commented:
@permcody There is a bit of a melt down with CIVET now and the stuff in next really needs to get in, so don't merge. If you approve I will merge when the storm ends.

lindsayad commented:
Hmm I wonder if the connective part just becomes too much for amg

> On Jun 14, 2019, at 5:26 PM, Andrew E Slaughter <notifications@github.com> wrote:
> 
> @aeslaughter commented on this pull request.
> 
> In tutorials/darcy_thermo_mech/step09_mechanics/problems/step9.i:
> 
> > @@ -217,6 +211,8 @@
>    steady_state_detection = true
>    dt = 0.25
>    solve_type = PJFNK
> +  automatic_scaling = true
> +  compute_scaling_once = false
> -pc_type lu works much better, it doesn't seem to get worse over time.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
